pkg/gocachemanager/sdk.go

```diff
@@ -111,13 +116,52 @@ func (gcw *GoCacheWrapper) Get(ctx context.Context, key []byte) ([]byte, error)
 		return nil, fmt.Errorf("getting data: %w", err)
 	}
 
-	return data, nil
+	var result []byte
+
+	result, err = base64.StdEncoding.DecodeString(value)
+	if err != nil {
+		return nil, fmt.Errorf("decoding cache value: %w", err)
+	}
+
+	if gcw.gzip {
+		gzipr, err := gzip.NewReader(bytes.NewReader(result))
+		if err != nil {
+			return nil, fmt.Errorf("decompressing cache value: %w", err)
+		}
+
+		result, err = io.ReadAll(gzipr)
+		if err != nil {
+			return nil, fmt.Errorf("reading gzip bytes: %w", err)
+		}
+
+		gzipr.Close()
+	}
```

Comment:
Sounds good, but think of the following scenario:

1. Started using cache-manager without gzip.
2. Many many values currently in the cache.
3. Read the docs and find out there's gzip.
4. Turn it on. Now my entire cache is invalid and I'll receive a ton of errors when I deploy the new version.

So, in order for the gzip to be safer, we need to think like a Browser :) If the content comes with gzip we process it with gzip, if we can't we process it normally. So gzip should not fail hard if it can't decode as it may well be a transition value (stored with `gzip: false`, read with `gzip:true`).

Please add tests that validate this as well? Both cases:

1. Store value with `gzip: false`, read with `gzip: true`
2. Store value with `gzip: true`, read with `gzip: false`

Both should work fine.

Reply:
The case 2 because when we try to unmarshalling a byte array changed by gzip a error is throw: "unmarshalling data: proto: cannot parse invalid wire-format data".